packages/deployments/contracts/contracts/messaging/RootManager.sol

```diff
@@ -14,122 +16,171 @@ import {WatcherClient} from "./WatcherClient.sol";
  * @notice This contract exists at cluster hubs, and aggregates all transfer roots from messaging
  * spokes into a single merkle root
  */
+contract RootManager is ProposedOwnable, IRootManager, WatcherClient, DomainIndexer {
+  // ============ Libraries ============
+
+  using QueueLib for QueueLib.Queue;
 
-contract RootManager is ProposedOwnable, IRootManager, WatcherClient {
   // ============ Events ============
 
   event RootAggregated(uint32 domain, bytes32 receivedRoot, uint256 index);
 
-  event RootPropagated(bytes32 aggregate, uint32[] domains);
+  event RootPropagated(bytes32 aggregate, uint32[] domains, uint256 count);
 
   event ConnectorAdded(uint32 domain, address connector);
 
   event ConnectorRemoved(uint32 domain, address connector);
```

Comment:
because the hashes stored in the indexer change, we should make sure we emit the newly updated domain and connector arrays